extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-base/src/main/java/org/apache/guacamole/auth/jdbc/sharing/connectiongroup/SharedRootConnectionGroup.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.guacamole.net.auth.ConnectionGroup;
+import org.apache.guacamole.net.auth.Directory;
+import org.apache.guacamole.protocol.GuacamoleClientInformation;
+
+/**
+ * A ConnectionGroup implementation which contains all connections accessible
+ * via a given SharedUserContext. The identifier of a SharedRootConnectionGroup
+ * is statically defined, and all Connections which are intended to be contained
+ * within an instance of SharedRootConnectionGroup MUST return that identifier
+ * via getParentIdentifier().
+ *
+ * @author Michael Jumper
+ */
+public class SharedRootConnectionGroup implements ConnectionGroup {
+
+    /**
+     * The identifier of the root connection group. All Connections which are
+     * intended to be contained within an instance of SharedRootConnectionGroup
+     * MUST return this identifier via getParentIdentifier().
+     */
+    public static final String IDENTIFIER = "ROOT";
+
+    /**
+     * The SharedUserContext through which this connection group is accessible.
+     */
+    private final SharedUserContext userContext;
+
+    /**
+     * Creates a new SharedRootConnectionGroup which contains all connections
+     * accessible via the given SharedUserContext. The SharedRootConnectionGroup
+     * is backed by the SharedUserContext, and any changes to the connections
+     * within the SharedUserContext are immediately reflected in the
+     * SharedRootConnectionGroup.
+     *
+     * @param userContext
+     *     The SharedUserContext which should back the new
+     *     SharedRootConnectionGroup.
+     */
+    public SharedRootConnectionGroup(SharedUserContext userContext) {
+        this.userContext = userContext;
+    }
+
+    @Override
+    public String getIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public void setIdentifier(String identifier) {
+        throw new UnsupportedOperationException("The root group is immutable.");
+    }
+
+    @Override
+    public String getName() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public void setName(String name) {
+        throw new UnsupportedOperationException("The root group is immutable.");
+    }
+
+    @Override
+    public String getParentIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public void setParentIdentifier(String parentIdentifier) {
+        throw new UnsupportedOperationException("The root group is immutable.");
+    }
+
+    @Override
+    public GuacamoleTunnel connect(GuacamoleClientInformation info)
+            throws GuacamoleException {
+        throw new GuacamoleSecurityException("Permission denied.");
+    }
+
+    @Override
+    public Map<String, String> getAttributes() {
+        return Collections.<String, String>emptyMap();
+    }
+
+    @Override
+    public void setAttributes(Map<String, String> attributes) {
+        // Do nothing - no attributes supported
+    }
+
+    @Override
+    public int getActiveConnections() {
+        return 0;
```

Comment:
Simply because the number of active connections shared connections by a particular user is not tracked. From the definition of `getActiveConnections()` within `Connectable`:

> Implementations may simply return 0 if this value is not tracked.